com.unity.shadergraph/Editor/Drawing/Inspector/InspectorView.cs

```diff
@@ -0,0 +1,219 @@
+﻿﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Drawing.Views;
+ using ICSharpCode.NRefactory.Ast;
+ using UnityEditor.Experimental.GraphView;
+ using UnityEditor.Graphing;
+ using UnityEditor.ShaderGraph.Drawing;
+ using UnityEngine.UIElements;
+
+ namespace Drawing.Inspector
+{
+    class InspectorView : GraphSubWindow
+    {
+        // References
+        readonly List<Type> m_PropertyDrawerList = new List<Type>();
+
+        // There's persistent data that is stored in the graph settings property drawer that we need to hold onto between interactions
+        private IPropertyDrawer m_graphSettingsPropertyDrawer = new GraphDataPropertyDrawer();
+
+        private Action m_previewUpdateDelegate;
+
+        public int currentlyDisplayedPropertyCount { get; private set; } = 0;
+
+        protected override string windowTitle => "Inspector";
+        protected override string elementName => "InspectorView";
+        protected override string styleName => "InspectorView";
+
+        static IEnumerable<Type> GetPropertyDrawerTypes(Assembly assembly)
+        {
+            foreach(Type type in assembly.GetTypes())
+            {
+                if (type.GetCustomAttributes(typeof(SGPropertyDrawer), true).Length > 0)
+                {
+                    yield return type;
+                }
+            }
+        }
+
+        public InspectorView(GraphView graphView, Action updatePreviewDelegate) : base(graphView)
+        {
+            this.m_previewUpdateDelegate = updatePreviewDelegate;
+
+            foreach (var assembly in AppDomain.CurrentDomain.GetAssemblies())

```

Comment:
Use [TypeCache](https://docs.unity3d.com/2020.1/Documentation/ScriptReference/TypeCache.html) if you can here.